nifi/Dockerfile

```diff
@@ -4,74 +4,51 @@
 # The source code is available at https://github.com/stackabletech/stackable-utils/tree/main/stackable-bcrypt
 # The binaries are published to Nexus with: mvn clean deploy
 # This jar used to be COPYed into the image from a local binary but it's not possible to do so anymore
 # because /bin is owned by root and curl cannot write there unless it runs as root.
 # This should be fixed by moving the jar to location under /stackable and updating the Nifi operator accordingly.
-RUN curl --fail -L 'https://repo.stackable.tech/repository/m2/tech/stackable/nifi/stackable-bcrypt/1.0-SNAPSHOT/stackable-bcrypt-1.0-20240508.153334-1-jar-with-dependencies.jar' -o /bin/stackable-bcrypt.jar \
-    && chown stackable:stackable /bin/stackable-bcrypt.jar
+RUN curl --fail -L 'https://repo.stackable.tech/repository/m2/tech/stackable/nifi/stackable-bcrypt/1.0-SNAPSHOT/stackable-bcrypt-1.0-20240508.153334-1-jar-with-dependencies.jar' \
+    # This used to be located in /bin/stackable-bcrypt.jar we create a softlink in the main container for backwards compatibility
+    -o /stackable/stackable-bcrypt.jar && \
+    chown stackable:stackable /stackable/stackable-bcrypt.jar
 
 USER stackable
 WORKDIR /stackable
 
-COPY --chown=stackable:stackable nifi/stackable /stackable
-COPY --chown=stackable:stackable nifi/licenses /licenses
-COPY --chown=stackable:stackable nifi/python /stackable/python
-
-
-# zip is different than tar and cannot be just piped, therefore the intermediate save and remove step to unzip
-RUN curl --fail -L https://repo.stackable.tech/repository/packages/nifi/nifi-${PRODUCT}-bin.zip -o /stackable/nifi-${PRODUCT}-bin.zip && \
-    unzip /stackable/nifi-${PRODUCT}-bin.zip && \
-    rm /stackable/nifi-${PRODUCT}-bin.zip && \
-    ln -s /stackable/nifi-${PRODUCT} /stackable/nifi
+RUN curl --fail -L "https://repo.stackable.tech/repository/packages/nifi/nifi-${PRODUCT}-source-release.zip" -o "/stackable/nifi-${PRODUCT}-source-release.zip" && \
+    unzip "nifi-${PRODUCT}-source-release.zip" && \
+    rm -rf "nifi-${PRODUCT}-source-release.zip" && \
+    mv nifi-${PRODUCT} nifi-${PRODUCT}-src && \
+    cd /stackable/nifi-${PRODUCT}-src/ && \
+    mvn clean install -DskipTests && \
+    mv /stackable/nifi-${PRODUCT}-src/nifi-assembly/target/nifi-${PRODUCT}-bin/nifi-${PRODUCT} /stackable/nifi-${PRODUCT} && \
+    rm -rf /stackable/nifi-${PRODUCT}-src 
 
 # Add Iceberg extensions as they are not included by default and are important enough
 # They need to be build from source, as https://mvnrepository.com/artifact/org.apache.nifi/nifi-iceberg-processors-nar does not ship the org.apache.hadoop.fs.s3a.S3AFileSystem (see https://github.com/apache/nifi/pull/6368#issuecomment-1502175258)
 # See https://repo.stackable.tech/repository/packages/nifi/iceberg-nars/README.md for details on how to build them
-RUN cd /stackable/nifi/lib/ && \
+RUN cd /stackable/nifi-${PRODUCT}/lib/ && \
```

Comment:
maybe these can be downloaded in `/stackable/libs `to not mingle with the source folder.

Reply:
Was copied in the binary folder before, just kept it like this.